crates/rocksdb/src/lib.rs

```diff
@@ -0,0 +1,242 @@
+    pub name: DbName,
+    pub owner: Owner,
+    pub path: PathBuf,
+    /// Options applied to the database _before_ applying RocksDbOptions loaded from disk/env.
+    pub db_options: rocksdb::Options,
+    /// Options of the column family are applied _before_ the values loaded from
+    /// RocksDbOptions from disk/env. Those act as default values for if the option
+    /// is defined in RockDbOptions.
+    ///
+    /// Overriding per-column family options from config file is not supported [yet].
+    pub column_families: Vec<(CfName, rocksdb::Options)>,
+    _phantom: std::marker::PhantomData<T>,
+}
+
+#[derive(derive_more::Display, Clone)]
+#[display(fmt = "{}::{}", owner, name)]
+pub struct RocksDb {
+    pub owner: Owner,
+    pub name: DbName,
+    pub db_options: rocksdb::Options,
+    pub column_families: BTreeMap<CfName, rocksdb::Options>,
+    db: Arc<dyn RocksAccess + Send + Sync + 'static>,
+}
+
+static_assertions::assert_impl_all!(RocksDb: Send, Sync);
+
+impl Deref for RocksDb {
+    type Target = Arc<dyn RocksAccess + Send + Sync + 'static>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.db
+    }
+}
+
+impl RocksDb {
+    pub(crate) fn new<T>(spec: DbSpec<T>, db: Arc<T>) -> Self
+    where
+        T: RocksAccess + Send + Sync + 'static,
+    {
+        Self {
+            owner: spec.owner,
+            name: spec.name,
+            db,
+            db_options: spec.db_options,
+            column_families: spec.column_families.into_iter().collect(),
+        }
+    }
+
+    /// Returns the raw rocksdb handle, this should only be used for server operations that
+    /// require direct access to rocksdb.
+    pub fn inner(&self) -> &Arc<dyn RocksAccess + Send + Sync + 'static> {
+        &self.db
+    }
+
+    pub fn cfs(&self) -> Vec<&CfName> {
+        self.column_families.keys().collect()
+    }
+
+    pub fn get_histogram_data(&self, histogram: Histogram) -> HistogramData {
+        self.db_options.get_histogram_data(histogram)
+    }
+
+    pub fn get_ticker_count(&self, ticker: Ticker) -> u64 {
+        self.db_options.get_ticker_count(ticker)
+    }
+
+    pub fn get_statistics_str(&self) -> Option<String> {
+        self.db_options.get_statistics()
+    }
+}
+
+pub(crate) fn amend_db_options(db_options: &mut rocksdb::Options, opts: &RocksDbOptions) {
+    db_options.create_if_missing(true);
+    db_options.create_missing_column_families(true);
+    db_options.set_atomic_flush(true);
+
+    if opts.rocksdb_num_threads() > 0 {
+        db_options.increase_parallelism(opts.rocksdb_num_threads() as i32);
+        db_options.set_max_background_jobs(opts.rocksdb_num_threads() as i32);
+    }
+
+    if !opts.rocksdb_disable_statistics() {
+        db_options.enable_statistics();
+        // Reasonable default, but we might expose this as a config in the future.
+        db_options.set_statistics_level(rocksdb::statistics::StatsLevel::ExceptDetailedTimers);
+    }
+
+    db_options.set_atomic_flush(true);
```

Comment:
This looks like a duplicate of line 166.

Reply:
I actually will remove this entirely, this should be a property of the specific database use-case.